File: sdk/ctv/tvos/Sources/CTVSDK/LoadError.swift
import Foundation

public enum LoadError: Equatable {
    case noFill
    case timeout
    case network
    case status(code: Int, label: String)
    case generic(_ reason: String)

    public var reason: String {
        switch self {
        case .noFill: return "no_fill"
        case .timeout: return "timeout"
        case .network: return "network_error"
        case let .status(_, label): return label
        case let .generic(reason): return reason
        }
    }
}

File: sdk/ctv/android-tv/src/main/kotlin/com/rivalapexmediation/ctv/network/LoadError.kt
package com.rivalapexmediation.ctv.network

sealed interface LoadError {
    object NoFill : LoadError
    object Timeout : LoadError
    object Network : LoadError
    data class Status(val label: String, val code: Int) : LoadError
    data class Generic(val reason: String) : LoadError
}

fun LoadError.reason(): String = when (this) {
    LoadError.NoFill -> "no_fill"
    LoadError.Timeout -> "timeout"
    LoadError.Network -> "network_error"
    is LoadError.Status -> label
    is LoadError.Generic -> reason
}
